### ISSUE TITLE
Proposal to drop support for old versions of SQL.

### ISSUE THREAD

**Issue by ItsAGeekThing** [StormlightTech/OpenCAD-php#384]
**Synopsis**
OpenCAD's officially supports MySQL >=5.5 (no recommended version) and MariaDB >=10.0 (with 10.3 being recommended) per our current documentation.
Running older versions of MySQL/MariaDB may expose servers to security vulnerabilities and children eating bugs that would have otherwise already been fixed on current releases. 

MySQL 5.5 (OpenCAD's minimum supported version) has reached [end of life](https://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf) (support ended on Dec. 2015).
MySQL 5.6's support ended in Feb. 2018 and MySQL 5.7's in Oct. 2020.

_The latest version of MySQL that currently has support is **8.0** (which ends in April 2023)._

MariaDB 10.0 (OpenCAD's minimum supported version) has reached [end of life](https://mariadb.org/about/#maintenance-policy) (support ended on Mar. 2019).
MariaDB 10.1's support ended in Oct. 2020.

_The latest version of MariaDB that currently has support is **10.2** (which ends in May 2022)_

**Where do we begin?**
We should set our minimal (and supported) version of MySQL as 8.0 (support end in April 2023).
We should set our minimal version of MariaDB as 10.2 (support end in May 2022) and the recommended version as 10.5 (support ends in June 2025).

**What do we do after?**
Once we shift OpenCAD's officially supported version of MySQL and MariaDB to maintained versions we should begin to follow [Oracle's Oracle Lifetime Support Policy](https://www.oracle.com/us/support/library/lifetime-support-technology-069183.pdf) and [MariaDB's Maintenance policy](https://mariadb.org/about/#maintenance-policy), respectively.

**What other considerations must we discuss before making any changes?**
While this is a minor change in principle, the fact is that there are so many different environments being utilized by our community. 

***

_* We base MySQL's EOL date off of its support end date and not the extended support date (Oracle charges a [support fee for extended support](https://www.oracle.com/support/lifetime-support/) which is intended for their enterprise customers)._

**Comment by Cambridgeport90** [comment]
That sounds sensible to me.

**Comment by mikespears** [comment]
Agreed, this needs to be done. No sense in supporting unsupported versions.

**Comment by Cambridgeport90** [comment]
That might work, in fact. Follow the mainstreamers LOL.

**Comment by phillf** [comment]
One the topic of SQL we need to check for both MySQL (proper) and MariaDB versions as the version strings are a unique format for each.

**Comment by devotedsouls** [comment]
This proposal has been approved by the OpenCAD team.

Issue will be updated when pull request is made for the documentation.
